### DISCUSSION TITLE
Some questions about solving gradients

### DISCUSSION THREAD

**Discussion by 15175829310** [mdolab/dafoam#687] [General]
May I use the finite difference method to calculate the gradient and validate the adjoint method? How should I operate？

**Comment by 15175829310** [comment]
I have read your previous replies to others. Do you have any cases related to flow field inversion

**Comment by friedenhe** [comment]
There is no automatic way to do FD for field inversion because there are so many inputs. What you can do is to manually perturb one beta value in the runScript.py file and run the primal to get the FD derivative. Check where you prescribe the initial values for beta in the runScript.py file. You can prescribe some perturbed beta values there.

**Reply by 15175829310** [reply]
Sorry, do you have any reference scripts? I'm not sure how to operate them

**Reply by 15175829310** [reply]
I saw an example of the periodic hill in your article before and would like to know how you wrote the script
“An Open-source Adjoint-based Field Inversion  Tool for Data-driven RANS Modelling”

**Comment by friedenhe** [comment]
https://github.com/DAFoam/tutorials/blob/607a4f7738d067dd0921eecd4a96750c69443811/Ramp/steady/train/runScript_FI.py#L201
Here is where you set the initial value for beta `val=np.ones(nCells)`, you can set a different initial value by manually perturbing beta at for certain cell.
